astronomer/airflow/version_check/update_checks.py

```diff
@@ -338,56 +350,130 @@ def __init__(self):
             # On runtime:
             # if the user is on version 5.0.6 and 5.0.8, 6.0.0 are available,
             # notify the user about 5.0.8 and don't notify user about 6.0.0.
             rel_parsed_version = version.parse(rel.version)
-
             rel_parsed_base_version = rel_parsed_version.major
 
             if rel_parsed_version > runtime_version and rel_parsed_base_version == base_version:
+                days_to_eol = None
+                if rel.end_of_support:
+                    now = utcnow()
+                    days_to_eol = (rel.end_of_support - now).days
                 return {
                     "level": rel.level,
                     "date_released": rel.date_released,
                     "description": rel.description,
                     "version": rel.version,
                     "url": rel.url,
                     "app_name": "Astronomer Runtime",
+                    "days_to_eol": days_to_eol,
+                    "dismissed_until": rel.eos_dismissed_until,
                 }
 
         if sorted_releases:
-            recent_release = sorted_releases[0]
+            # If no newer version is found, return the latest version within the same major version.
+            recent_release = next(
+                (rel for rel in sorted_releases if version.parse(rel.version).major == base_version),
+                sorted_releases[0],
+            )
+            days_to_eol = None
+            if recent_release.end_of_support:
+                now = utcnow()
+                days_to_eol = (recent_release.end_of_support - now).days
             return {
                 'level': recent_release.level,
                 'date_released': recent_release.date_released,
                 'description': recent_release.description,
                 'version': recent_release.version,
                 'url': recent_release.url,
                 "app_name": "Astronomer Runtime",
+                "days_to_eol": days_to_eol,
+                "dismissed_until": recent_release.eos_dismissed_until,
             }
+        return None
+
+    def get_eol_notice(self, current_version) -> dict[str, Any] | None:
+        """
+        Get the EOL notice information if the current version is near or past its EOL.
 
+        :param current_version: The current runtime version information.
+        """
+        if current_version and current_version.end_of_support:
+            now = utcnow()
+            days_to_eol = (current_version.end_of_support - now).days
+            eol_level = (
+                'critical'
+                if days_to_eol <= 0
+                else 'warning'
+                if days_to_eol <= self.eol_warning_threshold_days
+                else ''
+            )
+            return {
+                "level": eol_level,
+                "date_released": current_version.date_released,
+                "description": current_version.description,
+                "version": current_version.version,
```

Comment:
Similarly, for the EOL notices, we don't need the update notice stuff right?

Reply:
Removed it